edit.go

```diff
@@ -518,6 +517,40 @@ func (e *Edit) variables(v *vaulted.Vault) error {
 	return nil
 }
 
+func (e *Edit) setDuration(v *vaulted.Vault) {
+	var dur string
+	var err error
+	maxDuration := 36 * time.Hour
+	if v.AWSKey != nil && v.AWSKey.Role != "" {
+		maxDuration = time.Hour
+	}
+	readMessage := fmt.Sprintf("Duration (e.g. 15m or %v): ", parseDuration(maxDuration.String()))
+	dur, err = e.readValue(readMessage)
+	if err == nil {
+		duration, durErr := time.ParseDuration(dur)
+		if durErr != nil {
+			color.Red("%s", durErr)
+			return
+		}
+		if duration < 15*time.Minute || duration > maxDuration {
+			errorMessage := fmt.Sprintf("Duration must be between 15m and %v", parseDuration(maxDuration.String()))
+			color.Red(errorMessage)
+			return
+		}
+		v.Duration = duration
+	}
+}
+
+func parseDuration(dur string) string {
```

Comment:
This isn't actually parsing anything, so we should probably rename it. What if we called it something like `formatDuration` and took a `time.Time` as the parameter rather than a `string`? This way, all of the formatting and converting to a `string` happens inside this one function.